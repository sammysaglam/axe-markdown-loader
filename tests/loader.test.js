const parse = require('../src/parse').parse;
const buildComponent = require('../src/buildComponent').buildComponent;

const removeIndentationsBy4Tabs = string => string.replace(/^\s+/, '').replace(/\s+$/, '').replace(/\n\t\t\t\t/g, '\n');

/* eslint-disable no-undef */
describe('markdown loader', () => {
	describe('markdown parser', () => {

		it('correctly parses simple markdown', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				# Heading 1
				
				## Heading 2
				
				### Heading 3
				
				#### Heading 4
				
				##### Heading 5
				
				###### Heading 6
				
				Emphasis, aka italics, with *asterisks* or _underscores_.
				
				Strong emphasis, aka bold, with **asterisks** or __underscores__.
				
				Combined emphasis with **asterisks and _underscores_**.
				
				Strikethrough uses two tildes. ~~Scratch this.~~
				
				1. First ordered list item
				2. Another item
				⋅⋅* Unordered sub-list. 
				1. Actual numbers don't matter, just that it's a number
				⋅⋅1. Ordered sub-list
				4. And another item.
	
				⋅⋅⋅You can have properly indented paragraphs within list items. Notice the blank line above, and the leading spaces (at least one, but we'll use three here to also align the raw Markdown).
	
				⋅⋅⋅To have a line break without a paragraph, you will need to use two trailing spaces.⋅⋅
				⋅⋅⋅Note that this line is separate, but within the same paragraph.⋅⋅
				⋅⋅⋅(This is contrary to the typical GFM line break behaviour, where trailing spaces are not required.)
	
				* Unordered list can use asterisks
				- Or minuses
				+ Or pluses
				
				[I'm an inline-style link](https://www.google.com)
				
				[I'm an inline-style link with title](https://www.google.com "Google's Homepage")
				
				[I'm a reference-style link][Arbitrary case-insensitive reference text]
				
				[I'm a relative reference to a repository file](../blob/master/LICENSE)
				
				[You can use numbers for reference-style link definitions][1]
				
				Or leave it empty and use the [link text itself].
				
				URLs and URLs in angle brackets will automatically get turned into links. 
				http://www.example.com or <http://www.example.com> and sometimes 
				example.com (but not on Github, for example).
				
				Some text to show that the reference links can follow later.
				
				[arbitrary case-insensitive reference text]: https://www.mozilla.org
				[1]: http://slashdot.org
				[link text itself]: http://www.reddit.com
				
				Here's our logo (hover to see the title text):
	
				Inline-style: 
				![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 1")
				
				Reference-style: 
				![alt text][logo]
				
				[logo]: https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 2"
				
				Inline \`code\` has \`back-ticks around\` it.
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});
		});

		it('correctly parses js imports', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
			
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
			
				# Heading 1
				
				## Heading 2
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses additional attributes', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
					
				moreAttributes:
					'cool': 'test'
					'wow': 'man'
					
				watsup: 'man!'
					
				attribute-with-dashes: 'value'
					
				nestedAttributes:
					'cool':
						'test': 'i am nested'
						
					'wow':
						'deep nested':
							'i am': 'seriously nested'
				---
			
				# Heading 1
				
				## Heading 2
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses empty body with just imports & attributes', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
			
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
					
				moreAttributes:
					'cool': 'test'
					'wow': 'man'
					
				watsup: 'man!'
					
				attribute-with-dashes: 'value'
					
				nestedAttributes:
					'cool':
						'test': 'i am nested'
						
					'wow':
						'deep nested':
							'i am': 'seriously nested'
				---
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses JSX', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
			
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
				
				# Hello
				
				\`\`\`jsx
				<DefaultComponent
					name="Sammy"
					lastName="Saglam"
				/>
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses JSX and shows source', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
			
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
				
				# Hello
				
				\`\`\`jsx show-source
				<DefaultComponent
					name="Sammy"
					lastName="Saglam"
				/>
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses CSS and shows source', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
			
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
				
				# Hello
				
				\`\`\`css show-source
				
				body {
					background: red;
				}
				
				div .test > #incredible::before {
					content: 'cool';
					color: blue;
				}
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses SCSS (SASS) and shows source', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
				
				# Hello
				
				\`\`\`scss show-source
				
				body {
					background: red;
					
					$variable: #FFF;
					
					div .test > #incredible::before {
						content: 'cool';
						color: $variable;
					}
				}
				
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly parses markdown and inner fence blocks + shows source', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				# Writing markdown fence blocks

				\`\`\`md show-source
				# Title
				
				## The subtitle
				
				~~~css
				body {
				    background:red;
				}
				~~~
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

		it('correctly escapes backslashes in source code render', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				### Simple ORM
				\`\`\`php
				class Sale extends \\Axe\\ORM {
				   
				  public static function allowed_fields() {
				    return array(
				      "deliveryStatus",
				      "date",
				      "customerId",
				      "customerAddresses" => Address::get_allowed_fields(),
				      "productsSold"      => Product::get_allowed_fields()
				    );
				  }
				}
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => {

				try {
					expect(componentData).toMatchSnapshot();
					done();

				} catch (error) {
					done(error);
				}
			});

		});

	});

	describe('component builder', () => {

		it('correctly build component based on JSX source', done => {

			const markdownToTest = removeIndentationsBy4Tabs(`
		
				---
				imports:
					'{Compo , Apple , Pear}': '../src/PathToComponent'
					'DefaultComponent': 'module-name'
				---
				
				# Hello
				
				\`\`\`jsx show-source
				<DefaultComponent
					name="Sammy"
					lastName="Saglam"
				/>
				\`\`\`
				
			`);

			parse(markdownToTest).then(componentData => buildComponent(componentData)).then(component => {

				try {
					expect(component).toMatchSnapshot();

					done();

				} catch (error) {
					done(error);
				}
			});
		});

	});
});