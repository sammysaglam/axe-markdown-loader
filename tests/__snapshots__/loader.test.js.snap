// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown loader component builder correctly build component based on JSX source 1`] = `
"import React from 'react';
import ReactDOM from 'react-dom';
import reduce from 'reduce-object';

import 'prismjs';
import 'prismjs/components/prism-jsx';
import 'prismjs/components/prism-bash';
import 'prismjs/components/prism-scss';
import 'prismjs/components/prism-php';
import 'prismjs/components/prism-markdown';

const Remarkable = require('remarkable');
const md = new Remarkable({ xhtmlOut: true });

const highlightFenceBlock = (tokens, idx) => {

	const options = tokens[idx].params.split(/\\\\s+/g);
	const language = options[0];

	const { content } = tokens[idx];

	const prismLanguagesConverter = {
		js: 'javascript'
	};

	return \`<pre class=\\"language-\${language}\\"><code>\${Prism.highlight(content, Prism.languages[language])}</code></pre>\`;

};

md.renderer.rules.fence_custom.bash = highlightFenceBlock;
md.renderer.rules.fence_custom.json = highlightFenceBlock;
md.renderer.rules.fence_custom.html = highlightFenceBlock;
md.renderer.rules.fence_custom.jsx = highlightFenceBlock;
md.renderer.rules.fence_custom.javascript = highlightFenceBlock;
md.renderer.rules.fence_custom.js = highlightFenceBlock;
md.renderer.rules.fence_custom.css = highlightFenceBlock;
md.renderer.rules.fence_custom.scss = highlightFenceBlock;

import {Compo , Apple , Pear} from '../src/PathToComponent';
import DefaultComponent from 'module-name';
const importedComponents = {
	...({Compo , Apple , Pear}) ,
	DefaultComponent ,
};export default class MarkdownComponent extends React.Component {
	constructor(props) {
		super(props);

		this.addLineNumbers = this.addLineNumbers.bind(this);
	}

	componentDidMount() {
		window.addEventListener('resize', this.addLineNumbers);
		this.addLineNumbers();
	}

	componentWillUnmout() {
		window.removeEventListener('resize', this.addLineNumbers);
	}

	componentDidUpdate() {
		this.addLineNumbers();
	}

	addLineNumbers() {

		setTimeout(() => {

			if ( !this.rootElement ) return false;

			const NEW_LINE_EXP = /\\\\n(?!$)/g;

			const elements = [...this.rootElement.querySelectorAll('.axe-markdown__render-source--with-line-numbers')];

			elements.forEach(element => {

				const lines = element.textContent.split(NEW_LINE_EXP);
				const lineCount = (lines.length + 1) || 0;

				const newLineNumbersWrapper = document.createElement('span');
				newLineNumbersWrapper.setAttribute('aria-hidden', 'true');
				newLineNumbersWrapper.className = 'line-numbers-rows';
				newLineNumbersWrapper.innerHTML = new Array(lineCount).join('<span></span>');

				if (element.querySelectorAll('.line-numbers-rows').length === 0) {
					element.getElementsByTagName('code')[0].appendChild(newLineNumbersWrapper);
				}

				const lineNumbersWrapper = element.querySelector('.line-numbers-rows');

				if (element.querySelectorAll('.line-numbers-sizer').length === 0) {
					const newLineNumberSizer = document.createElement('span');
					newLineNumberSizer.className = 'line-numbers-sizer';
					element.getElementsByTagName('code')[0].appendChild(newLineNumberSizer);
				}

				const lineNumberSizer = element.querySelector('.line-numbers-sizer');
				lineNumberSizer.style.display = 'block';

				lines.forEach((line, lineNumber) => {

					lineNumberSizer.textContent = line || '\\\\n';
					const lineSize = lineNumberSizer.getBoundingClientRect().height;

					lineNumbersWrapper.children[lineNumber].style.height = lineSize + 'px';
				});

				lineNumberSizer.textContent = '';
				lineNumberSizer.style.display = 'none';
			});
		}, 1);
	}

	render() {
		const props = this.props;

		return (
			<div className=\\"axe-markdown\\" ref={rootElement => this.rootElement = rootElement}>
				<h1>Hello</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-jsx\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			<div className=\\"axe-markdown__render-result\\"> <DefaultComponent
   name=\\"Sammy\\"
   lastName=\\"Saglam\\"
/>
 </div>
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`<DefaultComponent
   name=\\"Sammy\\"
   lastName=\\"Saglam\\"
/>
\` ,
								Prism.languages.jsx
							)}}
						/>
					</pre>
				
		</div>
	
			</div>
		);
	}
}

const PropsTable = ({ component: { propTypes, defaultProps } }) => (
	<table className=\\"axe-markdown-props-table\\">
		<thead>
			<tr>
				<th>Prop Name</th>
				<th className=\\"axe-markdown-props-table__is-required\\">required?</th>
				<th>type</th>
				<th>Default</th>
			</tr>
		</thead>
		<tbody>
			{
				reduce(propTypes, (rows, { info }, propName) => [
					...rows,
					<tr key={propName}>
						<td>
							{propName}
							{
								console.log(info)
							}
						</td>
						<td className=\\"axe-markdown-props-table__is-required\\">
							{info.isRequired ? 'âœ…' : null}
						</td>
						<td>
							{readerFriendlyPropType({ info })}
						</td>
						<td className=\\"axe-markdown-props-table__default-props\\">
							{JSON.stringify(defaultProps[propName], null, 2)}
						</td>
					</tr>
				])
			}
		</tbody>
	</table>
);

const readerFriendlyPropType = ({ info }) => (

	info.propTypeName === 'array' && \`Array\`

	||

	info.propTypeName === 'bool' && \`Boolean\`

	||

	info.propTypeName === 'func' && \`Function\`

	||

	info.propTypeName === 'number' && \`Number\`

	||

	info.propTypeName === 'object' && \`Object\`

	||

	info.propTypeName === 'string' && \`String\`

	||

	info.propTypeName === 'symbol' && \`Symbol\`

	||

	info.propTypeName === 'node' && \`Number || String || React.Element || Array\`

	||

	info.propTypeName === 'element' && \`React.Element\`

	||

	info.propTypeName === 'instanceOf' && \`Instance of: \${info.className}\`

	||

	info.propTypeName === 'oneOf' && \`One of: (\${JSON.stringify(info.allowedValues, null, 3)})\`

	||

	info.propTypeName === 'oneOfType' && info.allowedPropTypes.map(({ info }) => (
		readerFriendlyPropType({ info })
	)).join(' || ')

	||

	info.propTypeName === 'arrayOf' && (
		<div>
			ArrayOf: ([
			<div className=\\"axe-markdown-props-table__inner\\">
				{
					readerFriendlyPropType({ info: info.allowedChildrenPropType.info })
				}
			</div>
			])
		</div>
	)

	||

	info.propTypeName === 'objectOf' && (
		<div>
			ArrayOf: ([
			<div className=\\"axe-markdown-props-table__inner\\">
				{
					readerFriendlyPropType({ info: info.allowedChildrenPropType.info })
				}
			</div>
			])
		</div>
	)

	||

	info.propTypeName === 'shape' && (
		<div>
			Object with Shape: (
			{'{'}
			{
				reduce(info.objectShape, (rows, {info}, key) => [
				...rows,
				<div className=\\"axe-markdown-props-table__inner\\" key={key}>
					<span className=\\"axe-markdown-props-table__object-shape-key\\">{key}</span>: (
					<div className=\\"axe-markdown-props-table__inner-next-line\\">
						{readerFriendlyPropType({ info })}
					</div>
				)</div>
				])
			}
			{'}'}
			)
		</div>
	)

	||

	info.propTypeName === 'any' && \`Any\`
);

class HtmlStyleTag extends React.Component {
	render() {

		const {content} = this.props;

		return ReactDOM.createPortal(
			<style dangerouslySetInnerHTML={{__html:content}}/> ,
			document.head
		);
	}
}

"
`;

exports[`markdown loader markdown parser correctly escapes backslashes in source code render 1`] = `
Object {
  "attributes": Object {},
  "jsx": "<h3>Simple ORM</h3>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-php\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`class Sale extends \\\\\\\\Axe\\\\\\\\ORM {
   
  public static function allowed_fields() {
    return array(
      \\"deliveryStatus\\",
      \\"date\\",
      \\"customerId\\",
      \\"customerAddresses\\" => Address::get_allowed_fields(),
      \\"productsSold\\"      => Product::get_allowed_fields()
    );
  }
}
\` ,
								Prism.languages.php
							)}}
						/>
					</pre>
				
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses CSS and shows source 1`] = `
Object {
  "attributes": Object {
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
  },
  "jsx": "<h1>Hello</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-css\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			<HtmlStyleTag content={\`
body {
   background: red;
}

div .test > #incredible::before {
   content: 'cool';
   color: blue;
}
\`} />
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`
body {
   background: red;
}

div .test > #incredible::before {
   content: 'cool';
   color: blue;
}
\` ,
								Prism.languages.css
							)}}
						/>
					</pre>
				
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses JSX 1`] = `
Object {
  "attributes": Object {
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
  },
  "jsx": "<h1>Hello</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-jsx\\",
				null
				
			].filter(className => className).join(' ')}
		>
			<div className=\\"axe-markdown__render-result\\"> <DefaultComponent
   name=\\"Sammy\\"
   lastName=\\"Saglam\\"
/>
 </div>
			
			
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses JSX and shows source 1`] = `
Object {
  "attributes": Object {
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
  },
  "jsx": "<h1>Hello</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-jsx\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			<div className=\\"axe-markdown__render-result\\"> <DefaultComponent
   name=\\"Sammy\\"
   lastName=\\"Saglam\\"
/>
 </div>
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`<DefaultComponent
   name=\\"Sammy\\"
   lastName=\\"Saglam\\"
/>
\` ,
								Prism.languages.jsx
							)}}
						/>
					</pre>
				
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses SCSS (SASS) and shows source 1`] = `
Object {
  "attributes": Object {
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
  },
  "jsx": "<h1>Hello</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-scss\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			<HtmlStyleTag content={\`body {
  background: red; }
  body div .test > #incredible::before {
    content: 'cool';
    color: #FFF; }
\`} />
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`
body {
   background: red;
   
   $variable: #FFF;
   
   div .test > #incredible::before {
      content: 'cool';
      color: $variable;
   }
}

\` ,
								Prism.languages.scss
							)}}
						/>
					</pre>
				
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses additional attributes 1`] = `
Object {
  "attributes": Object {
    "attribute-with-dashes": "value",
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
    "moreAttributes": Object {
      "cool": "test",
      "wow": "man",
    },
    "nestedAttributes": Object {
      "cool": Object {
        "test": "i am nested",
      },
      "wow": Object {
        "deep nested": Object {
          "i am": "seriously nested",
        },
      },
    },
    "watsup": "man!",
  },
  "jsx": "<h1>Heading 1</h1>
<h2>Heading 2</h2>
",
}
`;

exports[`markdown loader markdown parser correctly parses empty body with just imports & attributes 1`] = `
Object {
  "attributes": Object {
    "attribute-with-dashes": "value",
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
    "moreAttributes": Object {
      "cool": "test",
      "wow": "man",
    },
    "nestedAttributes": Object {
      "cool": Object {
        "test": "i am nested",
      },
      "wow": Object {
        "deep nested": Object {
          "i am": "seriously nested",
        },
      },
    },
    "watsup": "man!",
  },
  "jsx": "",
}
`;

exports[`markdown loader markdown parser correctly parses js imports 1`] = `
Object {
  "attributes": Object {
    "imports": Object {
      "DefaultComponent": "module-name",
      "{Compo , Apple , Pear}": "../src/PathToComponent",
    },
  },
  "jsx": "<h1>Heading 1</h1>
<h2>Heading 2</h2>
",
}
`;

exports[`markdown loader markdown parser correctly parses markdown and inner fence blocks + shows source 1`] = `
Object {
  "attributes": Object {},
  "jsx": "<h1>Writing markdown fence blocks</h1>

		<div
			className={[
				\\"axe-markdown__render\\",
				\\"axe-markdown__render--lang-markdown\\",
				\\"axe-markdown__render--with-source\\"
				
			].filter(className => className).join(' ')}
		>
			<div className=\\"axe-markdown__render-result\\" dangerouslySetInnerHTML={{__html:md.render(\`# Title

## The subtitle

\\\\\`\\\\\`\\\\\`css
body {
    background:red;
}
\\\\\`\\\\\`\\\\\`
\`)}} />
			
			
					<pre
						className={[
							\\"axe-markdown__render-source\\",
							\\"axe-markdown__render-source--with-line-numbers\\"
							
						].filter(className => className).join(' ')}
					>
						<code
							dangerouslySetInnerHTML={{__html:Prism.highlight(
								\`# Title

## The subtitle

\\\\\`\\\\\`\\\\\`css
body {
    background:red;
}
\\\\\`\\\\\`\\\\\`
\` ,
								Prism.languages.markdown
							)}}
						/>
					</pre>
				
		</div>
	",
}
`;

exports[`markdown loader markdown parser correctly parses simple markdown 1`] = `
Object {
  "attributes": Object {},
  "jsx": "<h1>Heading 1</h1>
<h2>Heading 2</h2>
<h3>Heading 3</h3>
<h4>Heading 4</h4>
<h5>Heading 5</h5>
<h6>Heading 6</h6>
<p>Emphasis, aka italics, with <em>asterisks</em> or <em>underscores</em>.</p>
<p>Strong emphasis, aka bold, with <strong>asterisks</strong> or <strong>underscores</strong>.</p>
<p>Combined emphasis with <strong>asterisks and <em>underscores</em></strong>.</p>
<p>Strikethrough uses two tildes. <del>Scratch this.</del></p>
<ol>
<li>First ordered list item</li>
<li>Another item
â‹…â‹…* Unordered sub-list.</li>
<li>Actual numbers don't matter, just that it's a number
â‹…â‹…1. Ordered sub-list</li>
<li>And another item.</li>
</ol>
<p>â‹…â‹…â‹…You can have properly indented paragraphs within list items. Notice the blank line above, and the leading spaces (at least one, but we'll use three here to also align the raw Markdown).</p>
<p>â‹…â‹…â‹…To have a line break without a paragraph, you will need to use two trailing spaces.â‹…â‹…
â‹…â‹…â‹…Note that this line is separate, but within the same paragraph.â‹…â‹…
â‹…â‹…â‹…(This is contrary to the typical GFM line break behaviour, where trailing spaces are not required.)</p>
<ul>
<li>Unordered list can use asterisks</li>
</ul>
<ul>
<li>Or minuses</li>
</ul>
<ul>
<li>Or pluses</li>
</ul>
<p><a href=\\"https://www.google.com\\">I'm an inline-style link</a></p>
<p><a href=\\"https://www.google.com\\" title=\\"Google's Homepage\\">I'm an inline-style link with title</a></p>
<p><a href=\\"https://www.mozilla.org\\">I'm a reference-style link</a></p>
<p><a href=\\"../blob/master/LICENSE\\">I'm a relative reference to a repository file</a></p>
<p><a href=\\"http://slashdot.org\\">You can use numbers for reference-style link definitions</a></p>
<p>Or leave it empty and use the <a href=\\"http://www.reddit.com\\">link text itself</a>.</p>
<p>URLs and URLs in angle brackets will automatically get turned into links.
http://www.example.com or <a href=\\"http://www.example.com\\">http://www.example.com</a> and sometimes
example.com (but not on Github, for example).</p>
<p>Some text to show that the reference links can follow later.</p>
<p>Here's our logo (hover to see the title text):</p>
<p>Inline-style:
<img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 1\\" /></p>
<p>Reference-style:
<img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 2\\" /></p>
<p>Inline <code>code</code> has <code>back-ticks around</code> it.</p>
",
}
`;
